dulwich/porcelain.py

```diff
@@ -170,17 +170,18 @@ def open_repo_closing(path_or_repo):
     return closing(Repo(path_or_repo))
 
 
-def path_to_tree_path(repopath, path):
+def path_to_tree_path(path):
     """Convert a path to a path usable in e.g. an index.
 
-    :param repo: Repository
-    :param path: A path
-    :return: A path formatted for use in e.g. an index
+    :param path: Path relative to repopath of a file in the repository
+    :return: A path formatted for use in e.g. an index (bytes)
     """
-    os.path.relpath(path, repopath)
+    if not isinstance(path, bytes):
+        path = path.encode(DEFAULT_ENCODING)
```

Comment:
You'll want to decode from sys.getfilesystemencoding() if original path was a filesystem path.

E.g. the filesystem path can be iso8859, while the git encoding is 'utf-8'.